lib/embulk/input/marketo/lead.rb

```diff
@@ -91,7 +91,12 @@ def self.generate_columns(metadata)
 
         def init
           @columns = task[:columns]
-          @ranges = task[:ranges][index]
+          if preview?
+            # Try earlier date at first to reduce cache miss hit
+            @ranges = task[:ranges].flatten.sort_by{|range| Time.parse(range["to"])}.reverse
```

Comment:
This line just re-order array to be newest date to oldest date.

Original `task[:ranges]` is a kind of such structure array.
```ruby
[
  {from: "2000-01-01", to: "2000-01-02"},
  {from: "2000-01-02", to: "2000-01-03"},
  ...
  {from: "2000-12-24", to: "2000-12-25"},
]
```

Re-order it as the following (new to old), and assign it as `@range` variable.
```ruby
[
  {from: "2000-12-24", to: "2000-12-25"},
  ...
  {from: "2000-01-02", to: "2000-01-03"},
  {from: "2000-01-01", to: "2000-01-02"},
]
```

When the preview is running, task worker try to get records with {from, to} values in the `@ranges` array.
`@ranges` first element is the newest {from, to} pair, so task worker will try the newest dates at first.